src/jsonpatcherproxy.js

```diff
@@ -257,13 +322,15 @@ var JSONPatcherProxy = (function() {
    * Revokes all proxies rendering the observed object useless and good for garbage collection @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/revocable}
    */
   JSONPatcherProxy.prototype.revoke = function() {
-    proxifiedObjectsMap.forEach(el => el.revoke());
+    this.proxifiedObjectsMap.forEach(el => {
+      el.proxy.revoke();
+    });
   };
   /**
    * Disables all proxies' traps, turning the observed object into a forward-proxy object, like a normal object that you can modify silently.
    */
   JSONPatcherProxy.prototype.disableTraps = function() {
-    proxifiedObjectsMap.forEach(this.disableTrapsForProxy, this);
```

Comment:
>Do we really need to create a new function every time?

Yes, we do now. Becuase `proxyInstance`, which I renamed to `revokableProxyInstance` has a property called originalObject this is necessary. I guess this [line](https://github.com/Palindrom/JSONPatcherProxy/blob/Fix-nested-array-deletion/src/jsonpatcherproxy.js#L209) makes it clear.